pkg/fs/dirname.go

```diff
@@ -0,0 +1,11 @@
+package fs
+
+import (
+	"path"
+	"runtime"
+)
+
+func Dirname() string {
+	_, filename, _, _ := runtime.Caller(1)
+	return path.Dir(filename)
+}
```

Comment:
```suggestion
import (
	"path"
	"os"
)

func exedir() {
	exe, err := os.Executable()
	if err != nil {
		panic(err)
	}
	
	return path.Dir(exe)
}

var Exedir = exedir()
```

Create and move to `pkg/test/helper_test.go`

Reply:
The paths obtained by runtime.Caller and os.Executable are not equivalent. What is expected in the test is to obtain the path to the test source code, so the implementation will not be modified here

ref

```go
_, Caller, _, _ := runtime.Caller(0)
fmt.Println("Caller", Caller) // app-builder/pkg/node-modules/cli_test.go

exe, _ := os.Executable()
fmt.Println("Executable", exe) //~/AppData\Local\Temp\go-build2984811955\b001\node-modules.test.exe
```

Reply:
Perhaps you need to add `// +build:ignore` in file header